src/passes/SignaturePruning.cpp

```diff
@@ -160,6 +167,10 @@ struct SignaturePruning : public Pass {
         continue;
       }
 
+      if (!subTypes.getStrictSubTypes(type).empty()) {
+        continue;
+      }
```

Comment:
Maybe update the comment below to cover both the supertype and subtype cases?

Reply:
Merged the comments.